ui/webui/resources/tools/build_webui.gni

```diff
@@ -0,0 +1,156 @@
+    }
+  }
+
+  # Move brave files to the place where chromium webui TS is compiled from.
+  # It should be the same directory that build_webui.gni uses after it performs
+  # its own preprocess_if_expr. Once chromium and brave frontend files are all in
+  # the same place, build_webui.gni will continue its typescript and rollup
+  # compile (etc.) from that directory.
+  if (_has_ts) {
+    preprocess_if_expr("brave_preprocess") {
+      if (defined(invoker.brave_preprocessor_defines)) {
+        defines = invoker.brave_preprocessor_defines
+      }
+      out_folder = "${target_gen_dir}/preprocessed"
+
+      in_files = brave_ts_files + brave_html_files
+
+      if (defined(invoker.brave_css_files)) {
+        in_files += invoker.brave_css_files
+      }
+    }
+  }
+
+  if (defined(invoker.brave_static_files)) {
+    extras_grdp = "$target_gen_dir/brave_extras.grdp"
+    generate_grd("brave_static_files") {
+      grd_prefix = "brave_extras"
+      out_grd = extras_grdp
+      input_files = invoker.brave_static_files
+      input_files_base_dir = invoker.brave_base_dir
+    }
+  }
+
+  build_webui(target_name) {
+    forward_variables_from(invoker,
+                           "*",
+                           [
+                             "brave_base_dir",
+                             "brave_static_files",
+                             "brave_web_component_files",
+                             "brave_non_web_component_files",
+                             "brave_css_files",
+                             "brave_preprocessor_defines",
+                             "brave_ts_definitions",
+                             "brave_mojo_files",
+                             "brave_mojo_files_deps",
+                             "brave_ts_deps",
+                           ])
+
+    if (defined(invoker.brave_static_files)) {
+      extra_grdp_deps = [ ":brave_static_files" ]
+      extra_grdp_files = [ extras_grdp ]
+    }
+
+    if (defined(invoker.brave_web_component_files)) {
+      web_component_files += invoker.brave_web_component_files
+    }
+
+    if (defined(invoker.brave_non_web_component_files)) {
+      non_web_component_files += invoker.brave_non_web_component_files
+    }
+
+    exclude_html_css_preprocess_files = brave_html_files
+
+    if (defined(invoker.brave_css_files)) {
+      css_files += invoker.brave_css_files
+      exclude_html_css_preprocess_files += invoker.brave_css_files
+    }
+
+    # make sure chromium doesn't try and move brave files from chromium src
+    # since we're taking care of that.
+    exclude_ts_preprocess_files = brave_ts_files
+    if (_has_ts) {
+      preprocess_deps = [ ":brave_preprocess" ]
+    }
+
+    # ts_extra_deps = brave_ts_extra_deps
+    if (defined(invoker.brave_ts_definitions)) {
+      ts_definitions += invoker.brave_ts_definitions
+    }
+
+    # if (optimize_webui) {
+    #   extra_grdp_deps += [ ":brave_preprocess" ]
+    #   extra_grdp_files = []
+    # }
```

Comment:
should this be uncommented? I think it'd be ideal if we can still output minified HTML/JS/CSS